test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs

```diff
@@ -25,51 +28,162 @@ namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
 {
     public class OtlpLogExporterTests : Http2UnencryptedSupportTests
     {
-        [Fact]
-        public void ToOtlpLogRecordTest()
-        {
-            // Just a basic test to demonstrate an
-            // approach useful for testing.
-            // This needs to be expanded to
-            // actually test the conversion.
-            List<LogRecord> logRecords = new List<LogRecord>();
+        private readonly ILogger logger;
+        private readonly List<LogRecord> exportedItems = new();
+        private readonly ILoggerFactory loggerFactory;
+        private OpenTelemetryLoggerOptions options;
 
-            using var loggerFactory = LoggerFactory.Create(builder =>
+        public OtlpLogExporterTests()
+        {
+            this.loggerFactory = LoggerFactory.Create(builder =>
```

Comment:
If you keep this option, then need to dispose : https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/OpenTelemetry.Tests/Logs/LogRecordTest.cs#L609-L612

OR, alternatively, don't do anything inside ctor. Create factory/logger inside each test (My preference is this way). This is a lot easier to test individual options.

Reply:
For example. if a particular test does `this.options.IncludeFormattedMessage = true;`, then it should "reverse" that in a finally block. (https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/OpenTelemetry.Tests/Logs/LogRecordTest.cs#L603-L606)

In short, I strongly encourage to create factory/logger in each unit tests :)

Reply:
Adopted the suggested approach, factor/logger in each unit tests! Thank you, Cijo!

Reply:
> In short, I strongly encourage to create factory/logger in each unit tests :)

+1. I personally would also prefer each test case to be very self-contained, rather than inventing some test framework and later turn it into a maintenance hell. Here goes what I did for Prometheus Exporter https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs.